### PR TITLE
Added support for dropdown type to tables in domain settings

### PR DESCRIPTION
**Pull request by seefo**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/672/HighFidelity-Beta-PR10683-049c27ce2fc4d823123e689d897963fe16056184.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/672/HighFidelity-Beta-PR10683-049c27ce2fc4d823123e689d897963fe16056184.dmg

File: domain-server/resources/web/settings/js/settings.js [hunk 1]
```diff
@@ -1435,6 +1451,12 @@ function addTableRow(row) {
           input.focus();
           focusChanged = true;
       }
+      
+      // if we are adding a dropdown, we should go ahead and make its select
+      // element is visible
+      if(isDropdown) {
```
Comment (birarda) [inline domain-server/resources/web/settings/js/settings.js]: space before `(isDropdown)`

File: domain-server/resources/web/settings/js/settings.js [hunk 2]
```diff
@@ -1416,11 +1428,15 @@ function addTableRow(row) {
...
         // with multiple we have an array of Objects, with one we have an array of whatever the value type is
         var num_columns = row.children('.' + Settings.DATA_COL_CLASS).length
+        var newName = setting_name + "[" + row_index + "]" + (num_columns > 1 ? "." + key : "");
 
         if (isCheckbox) {
-          input.attr("name", setting_name + "[" + row_index + "]" + (num_columns > 1 ? "." + key : ""))
+          input.attr("name", newName)
         } else {
-          input.attr("name", setting_name + "[" + row_index + "]" + (num_columns > 1 ? "." + key : ""))
+          if(isDropdown) {
```
Comment (birarda) [inline domain-server/resources/web/settings/js/settings.js]: space before `(isDropdown)`

File: domain-server/resources/web/settings/js/settings.js [hunk 3]
```diff
@@ -1281,6 +1281,17 @@ function makeTableHiddenInputs(setting, initialValues, categoryValue) {
           "<input type='checkbox' style='display: none;' class='form-control table-checkbox' " +
                  "name='" + col.name + "'" + (defaultValue ? " checked" : "") + "/>" +
         "</td>";
+    } else if (col.type === "select") {
+        html += "<td class='" + Settings.DATA_COL_CLASS + "'name='" + col.name + "'>"
+        html += "<select style='display: none;' class='form-control' data-hidden-input='" + col.name + "'>'"
+        
+        for(var i in col.options) {
```
Comment (birarda) [inline domain-server/resources/web/settings/js/settings.js]: space before (

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/682/HighFidelity-Beta-PR10683-2d9e92d1218bfa60184ffc176cff5348b3357864.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/682/HighFidelity-Beta-PR10683-2d9e92d1218bfa60184ffc176cff5348b3357864.dmg

**Pull request merged**